lightning/src/ln/router.rs

```diff
@@ -1049,6 +1085,28 @@ impl Router {
 
 		Err(LightningError{err: "Failed to find a path to the given destination", action: ErrorAction::IgnoreError})
 	}
+
+	/// Gets the list of announced channels, in random order. See ChannelInfo details field documentation for more information
+	pub fn list_edges(&self) -> Vec<ChannelInfo> {
```

Comment:
Oops, I didnt mean to expose the internal representation, I meant to expose the last_update_message field (if it is Some). Sadly, thats maybe not perfect here since we sometimes dont include it, but exposing the fields as-is is also not OK, since channel info is initialized with dummy values until we receive the first channel_updates - if we're gonna expose them they need to be Option<>al.